In [5]:
import pandas as pd
import numpy as np

# Load the dataset
data = pd.read_csv('AAL.csv')

# Convert the 'Date' column to datetime
data['Date'] = pd.to_datetime(data['Date'])

# Extract Year and Weekday from the Date column
data['Year'] = data['Date'].dt.year
data['Weekday'] = data['Date'].dt.day_name()

# Add a new column 'Return' to calculate daytime returns
data['Return'] = (data['Close'] - data['Open']) / data['Open']

# Initialize a dictionary to store results for each year
yearly_weekly_stats = {}

# Loop through each year
for year in data['Year'].unique():
    yearly_stats = {}
    
    # Filter the data for the current year
    year_data = data[data['Year'] == year]
    
    # Loop through each weekday to compute statistics
    for day in year_data['Weekday'].unique():
        day_data = year_data[year_data['Weekday'] == day]
        
        # All daytime returns (R)
        R = day_data['Return']
        
        # Negative returns (R−)
        R_neg = R[R < 0]
        
        # Non-negative returns (R+)
        R_pos = R[R >= 0]
        
        # Store statistics for the current day of the week
        yearly_stats[day] = {
            'μ(R)': R.mean(),
            'σ(R)': R.std(),
            '|R−|': len(R_neg),
            'μ(R−)': R_neg.mean() if len(R_neg) > 0 else np.nan,
            'σ(R−)': R_neg.std() if len(R_neg) > 0 else np.nan,
            '|R+|': len(R_pos),
            'μ(R+)': R_pos.mean() if len(R_pos) > 0 else np.nan,
            'σ(R+)': R_pos.std() if len(R_pos) > 0 else np.nan
        }
    
    # Add the statistics for this year to the overall dictionary
    yearly_weekly_stats[year] = yearly_stats

# Display the results for each year
for year, stats in yearly_weekly_stats.items():
    print(f"\nStatistics for the Year: {year}")
    stats_df = pd.DataFrame(stats).T
    print(stats_df)



Statistics for the Year: 2019
               μ(R)      σ(R)  |R−|     μ(R−)     σ(R−)  |R+|     μ(R+)  \
Wednesday  0.000222  0.018978  24.0 -0.015064  0.013881  27.0  0.013809   
Thursday  -0.001200  0.021968  26.0 -0.015676  0.018712  24.0  0.014482   
Friday    -0.000341  0.019508  25.0 -0.015659  0.011051  26.0  0.014388   
Monday     0.000098  0.017615  22.0 -0.014629  0.010320  26.0  0.012560   
Tuesday   -0.002525  0.019036  28.0 -0.016757  0.010654  24.0  0.014079   

              σ(R+)  
Wednesday  0.010712  
Thursday   0.012412  
Friday     0.013549  
Monday     0.011897  
Tuesday    0.011665  

Statistics for the Year: 2020
               μ(R)      σ(R)  |R−|     μ(R−)     σ(R−)  |R+|     μ(R+)  \
Thursday   0.001726  0.056787  25.0 -0.032563  0.029410  27.0  0.033476   
Friday    -0.012840  0.039367  32.0 -0.033388  0.027539  17.0  0.025839   
Monday     0.000765  0.058377  24.0 -0.036065  0.026369  24.0  0.037595   
Tuesday   -0.008417  0.048282  30.0 -0.038677  0.027543

3. Are there more days with negative or non-negative daytime returns?


To answer this question, we compare the values of |R−| (number of negative return days) and |R+| (number of non-negative return days) for each year and each weekday.

In most years and for most days of the week, the |R+| (non-negative days) is quite close to |R−| (negative days), but in some cases (like Fridays and Tuesdays), there are more negative return days.


For example, in 2020:


Tuesday: 30 negative days (|R−|) vs. 22 non-negative days (|R+|).


Friday: 32 negative days (|R−|) vs. 17 non-negative days (|R+|).


This suggests that certain days of the week tend to have more negative returns, especially Fridays in some years.


4. Does your stock lose more on a "down" day than it gains on an "up" day?


To answer this, we compare the values of μ(R−) (average loss on negative days) and μ(R+) (average gain on non-negative days).

For most of the years shown, the magnitude of μ(R−) (average loss) is larger than μ(R+) (average gain).


For example, in 2020 on Friday:


μ(R−) = -0.03388 (loss)
μ(R+) = 0.025839 (gain)


This indicates that the stock tends to lose more on down days than it gains on up days, implying that negative days are more pronounced in terms of loss than the positive days are in terms of gains.


5. Are these results the same across days of the week?


The results show some variability across days of the week:


Fridays often show a higher number of negative return days (|R−|) and more pronounced losses (μ(R−)).


Mondays and Wednesdays tend to have a more balanced distribution between negative and non-negative returns, with moderate values for both μ(R−) and μ(R+).


Thus, the behavior of the stock is not exactly the same across different days of the week. For instance, Fridays and some Tuesdays show stronger negative trends, while Mondays and Wednesdays appear to be more stable.

Conclusion:


Negative vs. Non-negative Days: There are more negative return days than non-negative ones, especially on certain weekdays like Fridays.


Loss vs. Gain: The stock tends to lose more on down days than it gains on up days, particularly on Fridays and Tuesdays.


Day of the Week Variability: The results vary by day of the week, with Fridays often being more negative compared to other days, while Mondays and Wednesdays are more balanced.

1. Are there any patterns across days of the week?


Yes, based on the tables for different years, we can observe some patterns across the days of the week:


Fridays tend to show more negative returns (higher |R−|) and often have larger average losses (μ(R−)) compared to the gains (μ(R+)) on positive days.


Example: In 2020, Friday had 32 negative days and an average loss of -0.03388, compared to only 17 non-negative days with an average gain of 0.025839.


Mondays tend to have more balanced returns, with fewer extreme losses and smaller deviations. In some years, Mondays show a small positive return on average.


Example: In 2019, Monday had an average return of 0.000098 with nearly equal negative and non-negative days.


Wednesdays also tend to be relatively stable, with fewer extreme losses compared to other days.


Example: In 2021, Wednesday had an average return of 0.001343, and the number of negative and non-negative days was nearly equal.

2. Are there any patterns across different years for the same day of the week?


>Yes, there are some recurring trends for certain days of the week across different years:

Fridays consistently show more negative returns across multiple years. This is visible in the data for 2019, 2020, 2021, and 2022, where Fridays have a larger number of negative days compared to other weekdays.


Tuesdays also tend to show a similar pattern, with more negative days in most years. For instance, in 2020, 2021, and 2022, Tuesdays show higher negative returns and larger losses on average.


Mondays are relatively stable across different years, often having a nearly balanced number of positive and negative days.

3. What are the best and worst days of the week to be invested for each year?


Based on the mean returns (μ(R)) and the number of positive and negative days, we can determine the best and worst days of the week for each year:

2019:


Best day: Monday (μ(R) = 0.000098) with balanced positive and negative returns.
Worst day: Friday (μ(R) = -0.000341) with more negative days (25 negative days).


2020:


Best day: Thursday (μ(R) = 0.001726) with relatively balanced positive and negative returns.
Worst day: Friday (μ(R) = -0.012840) with 32 negative days.


2021:


Best day: Monday (μ(R) = 0.002121) with a balanced return profile.
Worst day: Friday (μ(R) = -0.004395) with more negative days.


2022:


Best day: Tuesday (μ(R) = 0.003193), showing positive returns overall.
Worst day: Friday (μ(R) = -0.004261) with more negative days (29 negative days).


2023:


Best day: Monday (μ(R) = 0.002121) with balanced returns.
Worst day: Wednesday (μ(R) = -0.001470) with more negative days.

4. Do these days change from year to year for your stock?

   
Yes, the best and worst days to be invested change from year to year:


For example, Monday was the best day in 2019, 2021, and 2023, but in 2020, Thursday was the best day.


Similarly, Friday tends to be the worst day in most years, but in 2023, Wednesday showed more negative returns.

In [35]:
import pandas as pd
import numpy as np

# Load the dataset
data = pd.read_csv('AAL.csv')

# Convert the 'Date' column to datetime
data['Date'] = pd.to_datetime(data['Date'])

# Extract Year and Weekday from the Date column
data['Year'] = data['Date'].dt.year
data['Weekday'] = data['Date'].dt.day_name()

# Add a new column 'Return' to calculate daytime returns
data['Return'] = (data['Close'] - data['Open']) / data['Open']

# Initialize a dictionary to store results for each year
yearly_weekly_stats = {}

# Loop through each year
for year in data['Year'].unique():
    yearly_stats = {}
    
    # Filter the data for the current year
    year_data = data[data['Year'] == year]
    
    # Loop through each weekday to compute statistics
    for day in year_data['Weekday'].unique():
        day_data = year_data[year_data['Weekday'] == day]
        
        # All daytime returns (R)
        R = day_data['Return']
        
        # Negative returns (R−)
        R_neg = R[R < 0]
        
        # Non-negative returns (R+)
        R_pos = R[R >= 0]
        
        # Store statistics for the current day of the week
        yearly_stats[day] = {
            'μ(R)': R.mean(),
            'σ(R)': R.std(),
            '|R−|': len(R_neg),
            'μ(R−)': R_neg.mean() if len(R_neg) > 0 else np.nan,
            'σ(R−)': R_neg.std() if len(R_neg) > 0 else np.nan,
            '|R+|': len(R_pos),
            'μ(R+)': R_pos.mean() if len(R_pos) > 0 else np.nan,
            'σ(R+)': R_pos.std() if len(R_pos) > 0 else np.nan
        }
    
    # Add the statistics for this year to the overall dictionary
    yearly_weekly_stats[year] = yearly_stats

# To store the aggregate statistics for 5 years
aggregate_stats = {}

# Loop through each weekday and compute aggregate mean and std across 5 years
for day in data['Weekday'].unique():
    mu_r_list = []
    sigma_r_list = []
    
    # Collect the mean and standard deviation for each year
    for year, stats in yearly_weekly_stats.items():
        if day in stats:
            mu_r_list.append(stats[day]['μ(R)'])
            sigma_r_list.append(stats[day]['σ(R)'])
    
    # Calculate the aggregate mean and std
    aggregate_stats[day] = {
        'Aggregate μ(R)': np.mean(mu_r_list),
        'Aggregate σ(R)': np.mean(sigma_r_list)
    }

# Convert aggregate statistics to DataFrame
aggregate_stats_df = pd.DataFrame(aggregate_stats).T

# Display aggregate statistics
print("Aggregate Statistics Across All 5 Years:")
print(aggregate_stats_df)

# Determine the best and worst days based on the aggregate mean return
best_day = aggregate_stats_df['Aggregate μ(R)'].idxmax()
worst_day = aggregate_stats_df['Aggregate μ(R)'].idxmin()

print(f"\nBest day to invest: {best_day} with average μ(R) = {aggregate_stats_df['Aggregate μ(R)'].max():.6f}")
print(f"Worst day to invest: {worst_day} with average μ(R) = {aggregate_stats_df['Aggregate μ(R)'].min():.6f}")

# Step 2: Calculate how many days fall outside the μ ± 2σ range
# Calculate the overall mean (μ) and std (σ) for the entire dataset
overall_mean = data['Return'].mean()
overall_std = data['Return'].std()

# Calculate μ - 2σ and μ + 2σ
lower_bound = overall_mean - 2 * overall_std
upper_bound = overall_mean + 2 * overall_std

# Find the number of days where returns are outside μ ± 2σ
outliers = data[(data['Return'] < lower_bound) | (data['Return'] > upper_bound)]

# Number of outlier days
num_outliers = len(outliers)

# Total trading days
total_days = len(data)

# Expected number of outliers based on a normal distribution (5% of total days)
expected_outliers = total_days * 0.05

# Display the results
print(f"\nTotal trading days: {total_days}")
print(f"Number of trading days outside μ ± 2σ: {num_outliers}")
print(f"Expected number of days outside μ ± 2σ (5%): {round(expected_outliers)}")




Aggregate Statistics Across All 5 Years:
           Aggregate μ(R)  Aggregate σ(R)
Wednesday       -0.001416        0.027916
Thursday        -0.001930        0.032333
Friday          -0.004170        0.026229
Monday           0.000588        0.029436
Tuesday         -0.001497        0.027827

Best day to invest: Monday with average μ(R) = 0.000588
Worst day to invest: Friday with average μ(R) = -0.004170

Total trading days: 1258
Number of trading days outside μ ± 2σ: 54
Expected number of days outside μ ± 2σ (5%): 63


Is It Consistent with Normality?


Actual Outliers: 54 days (around 4.29% of the total days) fall outside the μ ± 2σ range.


Expected Outliers: Based on the assumption of normality, we expect 62.9 days (5% of the total days) to be outliers.


Conclusion:


The observed number of outlier days (54) is close to the expected number (62.9), but slightly lower.


This small difference suggests that the returns are fairly consistent with normality, though the distribution might not perfectly match a normal distribution. In practice, financial returns often exhibit fat tails or higher kurtosis, meaning that extreme values (outliers) can occur more frequently than in a normal distribution.


In this case, however, the results show that stock's daytime returns are reasonably close to a normal distribution, though slightly fewer extreme days were observed than expected.

In [40]:
data = pd.read_csv('AAL.csv')

# Convert the 'Date' column to datetime
data['Date'] = pd.to_datetime(data['Date'])

# Sort the data by date just in case it's not sorted
data = data.sort_values('Date')

# Calculate daytime returns (as done previously)
data['Return'] = (data['Close'] - data['Open']) / data['Open']

# Question 4, Part 1: Simulate for p = 1 (oracle is always correct)
starting_balance = 100  # Starting with $100
balance = starting_balance

# Oracle is always correct, so we apply all positive returns (because p = 1)
for ret in data['Return']:
    if ret > 0:  # Only make money on positive return days
        balance *= (1 + ret)

# Final balance after 5 years of trading with oracle
print(f"Final balance after 5 years with p = 1: ${balance:.2f}")

# Question 4, Part 2: How many days to reach the desired amount
# Last 3 digits of BU ID (782 as per your BU ID)
bu_id_last_3_digits = 782  # Your BU ID
desired_amount = starting_balance + bu_id_last_3_digits

balance = starting_balance
days = 0

# Loop over the returns and track the number of days to reach the desired amount
for ret in data['Return']:
    if ret > 0:  # Oracle is correct, apply positive returns
        balance *= (1 + ret)
    days += 1
    if balance >= desired_amount:
        break  # Stop once we've reached the desired amount

# Print how many days it took to reach the desired amount
print(f"Number of days to reach ${desired_amount}: {days} trading days")


Final balance after 5 years with p = 1: $17140124.67
Number of days to reach $882: 299 trading days


### 1)

In [49]:
data = pd.read_csv('AAL.csv')

# Convert the 'Date' column to datetime
data['Date'] = pd.to_datetime(data['Date'])

# Sort the data by date just in case it's not sorted
data = data.sort_values('Date')

# Calculate daytime returns for buy-and-hold strategy (Close - Open) / Open
data['Return'] = (data['Close'] - data['Open']) / data['Open']

### Part 1: Final Balance after 5 Years of Buy-and-Hold ###
starting_balance = 100  # Start with $100
balance = starting_balance

# Apply buy-and-hold for each trading day
for ret in data['Return']:
    balance *= (1 + ret)

# Final balance after 5 years of trading with buy-and-hold strategy
print(f"Final balance after 5 years (Buy-and-Hold): ${balance:.2f}")


Final balance after 5 years (Buy-and-Hold): $6.45


### 2

The results of the buy-and-hold strategy in Question 5 show a significantly lower final balance compared to the results obtained in Question 4 with the oracle strategy.


In Question 4, where the oracle had perfect predictions with 𝑝=1


p=1, the final balance reached an impressive $17,140,124.67 after 5 years. This is because the oracle only allowed us to invest on days with positive returns, completely avoiding losses.


In contrast, in Question 5, using the buy-and-hold strategy, the final balance after 5 years was just $6.45. This strategy exposed the investment to both positive and negative returns, and over time, the negative returns outweighed the positive ones, causing the balance to decline substantially.


In summary, the oracle strategy in Question 4 resulted in exponential growth due to avoiding negative returns, whereas the buy-and-hold strategy in Question 5 led to a sharp decrease in balance due to the compounding of both gains and losses.

### 3

In [57]:
winter_vacation_data = data[~data['Date'].dt.month.isin([12, 1, 2])]

# Reset balance and simulate buy-and-hold without trading in Dec, Jan, and Feb
balance = starting_balance
for ret in winter_vacation_data['Return']:
    balance *= (1 + ret)

print(f"Final balance with Winter Vacation (No trading in Dec, Jan, Feb): ${balance:.2f}")


Final balance with Winter Vacation (No trading in Dec, Jan, Feb): $6.31


Based on the final balance of 6.31 dollars, we can conclude that taking a winter vacation does not significantly improve the outcome compared to the regular buy-and-hold strategy, which resulted in a final balance of 6.45 dollars..

Key Observations:


The difference between the regular buy-and-hold strategy (6.45 dollars) and the winter vacation strategy (6.31 dollars) is minimal, indicating that avoiding trading during December, January, and February had little impact on the overall results.


Both strategies resulted in a significant reduction from the initial $100, showing that the negative returns throughout the year had a stronger impact on the investment, and skipping the winter months did not offer much protection from losses.

In [72]:
# Dictionary to map month numbers to month names
month_names = {
    1: "January", 2: "February", 3: "March", 4: "April", 
    5: "May", 6: "June", 7: "July", 8: "August", 
    9: "September", 10: "October", 11: "November", 12: "December"
}

# Calculate the balance for regular buy-and-hold (no skipping any month)
balance = starting_balance
for ret in data['Return']:
    balance *= (1 + ret)

# Add the Daytime Buy-and-Hold (B&H) strategy to the results
print(f"Final balance for Daytime Buy-and-Hold (B&H): ${balance:.2f}")
results = [("Daytime Buy-and-Hold (B&H)", balance)]

# Function to calculate final balance when skipping a particular month
def calculate_balance_skipping_month(skip_month):
    balance = starting_balance
    # Filter out the rows for the given month
    month_vacation_data = data[~(data['Date'].dt.month == skip_month)]
    for ret in month_vacation_data['Return']:
        balance *= (1 + ret)
    return balance

# Loop over all months from January (1) to December (12)
for month in range(1, 13):
    final_balance = calculate_balance_skipping_month(month)
    month_name = month_names[month]
    results.append((month_name, final_balance))
    print(f"Final balance skipping {month_name}: ${final_balance:.2f}")

# Find the best and worst months to take a vacation
best_month = max(results[1:], key=lambda x: x[1])  # Skip the first result (B&H)
worst_month = min(results[1:], key=lambda x: x[1])  # Skip the first result (B&H)

print(f"\nBest month to take a vacation: {best_month[0]} with final balance: ${best_month[1]:.2f}")
print(f"Worst month to take a vacation: {worst_month[0]} with final balance: ${worst_month[1]:.2f}")



Final balance for Daytime Buy-and-Hold (B&H): $6.45
Final balance skipping January: $6.02
Final balance skipping February: $6.11
Final balance skipping March: $14.20
Final balance skipping April: $13.17
Final balance skipping May: $8.07
Final balance skipping June: $6.27
Final balance skipping July: $7.59
Final balance skipping August: $8.71
Final balance skipping September: $7.32
Final balance skipping October: $7.79
Final balance skipping November: $8.61
Final balance skipping December: $7.14

Best month to take a vacation: March with final balance: $14.20
Worst month to take a vacation: January with final balance: $6.02


In [109]:
import random
import pandas as pd

# Function to simulate the portfolio value based on p
def simulate_portfolio_with_oracle(p, data):
    balance = 100  # Start with $100
    for ret in data['Return']:
        # Generate a random number between 0 and 1
        r = random.random()
        # If r <= p, oracle gives the correct label (positive return is used)
        # Otherwise, the opposite label is used (negative return is used)
        if (r <= p and ret >= 0) or (r > p and ret < 0):
            balance *= (1 + ret)
    return balance

# Daytime buy-and-hold without Oracle
balance_bh = 100
for ret in data['Return']:
    balance_bh *= (1 + ret)

print(f"Final balance for Daytime Buy-and-Hold: ${balance_bh:.2f}")

# Dictionary to store results for different values of p
results = [("Daytime Buy-and-Hold", balance_bh)]

# List of p values to test
p_values = [i / 10 for i in range(11)]  # 0, 0.1, 0.2, ..., 1.0

# Simulate for each p value
for p in p_values:
    final_balance = simulate_portfolio_with_oracle(p, data)
    results.append((f"p = {p}", final_balance))
    print(f"Final balance with p = {p}: ${final_balance:.2f}")



Final balance for Daytime Buy-and-Hold: $6.45
Final balance with p = 0.0: $0.00
Final balance with p = 0.1: $0.00
Final balance with p = 0.2: $0.01
Final balance with p = 0.3: $0.14
Final balance with p = 0.4: $2.03
Final balance with p = 0.5: $40.37
Final balance with p = 0.6: $266.81
Final balance with p = 0.7: $3158.04
Final balance with p = 0.8: $80254.20
Final balance with p = 0.9: $846076.41
Final balance with p = 1.0: $17140124.67


Value of p= 4 (2.76) is closer to buy and hold than 

In [117]:
import pandas as pd

# Assuming `data` is the dataset with 'Return' column

# Sort returns to find the best and worst trading days
sorted_returns_desc = data['Return'].sort_values(ascending=False)  # Best days at the top
sorted_returns_asc = data['Return'].sort_values(ascending=True)    # Worst days at the top

# Scenario a: Miss the 10 best trading days
def scenario_a(data):
    # Remove the 10 best trading days
    remaining_returns = sorted_returns_desc[10:]  # Remove top 10 days
    balance = 100
    for ret in remaining_returns:
        balance *= (1 + ret)
    return balance

# Scenario b: Invest in the 10 worst trading days
def scenario_b(data):
    # Only include the 10 worst trading days
    worst_10_days = sorted_returns_asc[:10]  # Take the worst 10 days
    balance = 100
    for ret in worst_10_days:
        balance *= (1 + ret)
    return balance

# Scenario c: Miss the 5 best trading days and invest in the 5 worst trading days
def scenario_c(data):
    # Remove the top 5 best days and include the worst 5 days
    remaining_returns = pd.concat([sorted_returns_desc[5:], sorted_returns_asc[:5]])
    balance = 100
    for ret in remaining_returns:
        balance *= (1 + ret)
    return balance

# Daytime Buy-and-Hold Strategy (B&H)
def daytime_buy_and_hold(data):
    balance = 100
    for ret in data['Return']:
        balance *= (1 + ret)
    return balance

# p = 1 (Never lose, perfect oracle)
def perfect_oracle(data):
    balance = 100
    for ret in data['Return']:
        if ret >= 0:
            balance *= (1 + ret)  # Only positive returns are considered
    return balance

# Calculate the balances for each strategy
buy_and_hold_balance = daytime_buy_and_hold(data)
perfect_oracle_balance = perfect_oracle(data)
miss_10_best_balance = scenario_a(data)
invest_10_worst_balance = scenario_b(data)
miss_5_best_invest_5_worst_balance = scenario_c(data)

# Print the results in a table format
print("Strategy                                 | Stock")
print("------------------------------------------------------")
print(f"Buy-and-Hold (B&H):                      | ${buy_and_hold_balance:.2f}")
print(f"Daytime Buy-and-Hold (B&H):              | ${buy_and_hold_balance:.2f}")
print(f"p = 1 (Never lose):                      | ${perfect_oracle_balance:.2f}")
print(f"Miss 10 best daytimes:                   | ${miss_10_best_balance:.2f}")
print(f"Invested in 10 worst daytimes:           | ${invest_10_worst_balance:.2f}")
print(f"Miss 5 best and invested in 5 worst:     | ${miss_5_best_invest_5_worst_balance:.2f}")


Strategy                                 | Stock
------------------------------------------------------
Buy-and-Hold (B&H):                      | $6.45
Daytime Buy-and-Hold (B&H):              | $6.45
p = 1 (Never lose):                      | $17140124.67
Miss 10 best daytimes:                   | $1.85
Invested in 10 worst daytimes:           | $30.67
Miss 5 best and invested in 5 worst:     | $1.41


### 1

Scenario (a): Missed the 10 Best Daytimes

The final balance after missing the 10 best trading days is $1.85. This shows a significant decrease from the buy-and-hold strategy because the most profitable days were missed.


Scenario (b): Invested in the 10 Worst Daytimes

The final balance after investing in the 10 worst trading days is $30.67. Surprisingly, this shows a lesser decrease than missing the 10 best days, which implies that the worst days weren’t as damaging compared to the profits of the best days.


Scenario (c): Missed the 5 Best Daytimes and Invested in the 5 Worst Daytimes


The final balance after missing the 5 best days and being invested in the 5 worst days is $1.41. This is even worse than missing the 10 best days alone, reflecting the combined negative impact of missing the best days and being invested in the worst days.

### 2

I lose more by missing the best daytimes compared to being invested on the worst daytimes. This is evident from the balances:
Missing the best 10 daytimes: $1.85.


Investing in the worst 10 daytimes: $30.67.


Missing the most profitable days has a greater impact on your overall balance than being invested in the least profitable days.

### 3


p if you used a statistical oracle instead of an angry one:
For Scenario (a) (missing the best 10 daytimes), the probability p could be estimated as 0.7 to 0.8, because the statistical oracle would likely predict positive returns on most days, but missing the top-performing days would still affect the balance.


For Scenario (b) (investing in the worst 10 daytimes), the probability p could be 0.5 to 0.6, as the statistical oracle would still correctly predict many positive returns, but occasionally allow investment on negative-return days.


For Scenario (c) (missing 5 best and investing in 5 worst daytimes), the probability p could be 0.5, reflecting a roughly even mix of good and bad predictions.
